test/lib/test_runner.go

```diff
@@ -136,6 +140,17 @@ func TearDown(client *Client) {
 			client.T.Logf("EVENT: %v", e)
 		}
 	}
+
+	// If the test is run by CI, export the pod logs in the namespace to the artifacts directory,
+	// which will then be uploaded to GCS after the test job finishes.
+	if prow.IsCI() {
```

Comment:
Does the test runner pod need permission to write to GCS, or is the GCS upload handled by something else?

Reply:
The GCS upload is done by a sidercar container that is running in the Prow job pod, after the Prow job is finished. It will use the service account we use in running Prow jobs as the credential, which has the write permission to the GCS bucket.